modules/hello-world/examples/DocParser.java

```diff
@@ -0,0 +1,338 @@
+					"import org.springframework.beans.factory.annotation.Autowired;\n" +
+					"import org.springframework.stereotype.Service;\n" +
+					"\n" +
+					"public class ${CLASSNAME} {\n" +
+					"\n" +
+					"  private static final String FLIGHTS_COLLECTION_NAME = \"flightsCollection\";\n" +
+					"  private static final String USERS_COLLECTION_NAME = \"usersCollection\";\n" +
+					"\n" +
+					"  String connectionString=\"localhost\";\n" +
+					"  String username=\"Administrator\";\n" +
+					"  String password=\"password\";\n" +
+					"  String bucketName=\"travel-sample\";\n" +
+
+					"  String id = \"user::\" + UUID.randomUUID().toString();\n" +
+					"  DurabilityLevel expiry;\n" +
+					"\n" +
+					"  Cluster cluster;\n" +
+					"  Bucket bucket;\n" +
+					"  Scope scope;\n" +
+					"  Collection collection;\n" +
+					"\n" +
+					"  BucketSettings bucketSettings;\n" +
+					"\n" +
+					"  CouchbaseMap map;\n" +
+					"  CouchbaseArrayList arrayList;\n" +
+					"  CouchbaseArraySet arraySet;\n" +
+					"  CouchbaseQueue queue;\t" +
+					"\n" +
+					"  private void init(){\n" +
+					"    ClusterEnvironment environment = ClusterEnvironment.builder().build();\n" +
+					"    cluster = Cluster.connect(connectionString,\n" +
+					"      ClusterOptions.clusterOptions(username, password).environment(environment));\n" +
+					"    bucket = cluster.bucket(\"travel-sample\");\n" +
+					"    scope = bucket.defaultScope();\n" +
+					"    collection = bucket.defaultCollection();\n" +
+					"  }\n";
+
+	String postamble =  "}\n";
+
+	private BufferedWriter openCodeFileIfNecessary(String dirName, String classname) throws IOException {
+		if(codeWriter == null) {
+			File dir=new File(dirName);
+			if(!dir.exists()){
+				dir.mkdir();
+			}
+			codeWriter = new BufferedWriter(new FileWriter(dirName + "/" + classname + ".java"));
+			codeWriter.write(preamble.replace("${CLASSNAME}", classname));
+		}
+		return codeWriter;
+	}
+
+	private void closeCodeFile() throws IOException {
+		  if(codeWriter != null) {
+				codeWriter.write(postamble);
+				codeWriter.close();
+				codeWriter = null;
+			}
+	}
+
+/*
+ * For each file
+ * 1) create a new *.adoc file containing the in-line [source,java] replaced with an include
+ * 2) create a new *.java file containing a method for each [source,java]
+ *
+ */
+	public void process(String fileName) throws IOException {
+		fileNo++;
+		int lineno = 0;
+		int sourceSection = 0;
+		Queue<String> adoc = new LinkedList<>();
+		String dirName = fileName.substring(0, fileName.lastIndexOf("/"));
+		String baseName = fileName.substring(fileName.lastIndexOf("/")+1, fileName.lastIndexOf(".adoc"));
+		String javaName = baseName
+				.replace("-", "_")
+				.replace(".", "_");
+		String javaFilename = dirName+"/../examples" + "/" + javaName +".java";
+		File javaFile= new File(javaFilename);
+		if ( javaFile.exists()){
+			System.out.println("java file already exists: "+javaFile.getPath());
```

Comment:
Should the exiting file be deleted and recreated?

Reply:
I manually edit the files after they are created (remove unnecessary imports etc). So I only want them generated if they don't already exists. ( I have manually renamed the files to be more java-class-like names, so this isn't an issue).